BaseBall/BaseBallTest/domain/RefereeTest.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  RefereeTest.swift
+//  BaseBallTest
+//
+//  Created by KimMinSeok on 4/8/24.
+//
+
+import XCTest
+
+final class RefereeTest: XCTestCase {
+
+    func test_Referee는_볼을_출력한다(){
+        let number1 = [1,2,3]
+        let number2 = [3,4,5]
+        
+        let Referee = Referee()
+        XCTAssertEqual(Referee.ballCount(inputNumbers1: number1, inputNumbers2: number2), 1)
```

Comment:
given, when, then 을 사용하면 가독성을 증가시킬 수 있습니다:)
[참고](https://www.swiftbysundell.com/articles/making-swift-tests-easier-to-debug/)

```suggestion
        // given
        let number1 = [1,2,3] 
        let number2 = [3,4,5]
        let referee = Referee()

        // when 
        let actual = referee.ballCount(inputNumbers1: number1, inputNumbers2: number2)
        
        // then
        XCTAssertEqual(actual, 1)
```

Reply:
3b6556425430addf29b138e95ec977c8262fd165 
given-when-then을 배웠는데 따로 TDD로 하면서 다 지우고 시작을 해서 까먹고 안해버렸네요!!

이제부터는 given-when-then을 모든 테스트코드에 적고 시작해야겠네요!

그리고 아래는 어떻게 given-when-then을 진행하면 좋을까요?
```swift
    func test_3자리의_수가_들어와야_한다() throws {
        XCTAssertThrowsError(try UserBall(numbers: [1,2,3,4]))
    }
    
    func test_숫자는_같은_수가_입력되면_안된다() throws {
        XCTAssertThrowsError(try UserBall(numbers: [3,3,1]))
    }
    
    func test_숫자는_0은_불가능하다() throws {
        XCTAssertThrowsError(try UserBall(numbers: [3,2,0]))
    }
```

Reply:
numbers가 given 이고 when은 비워야 하나...then에 XCTAssertThrowsError(try UserBall(numbers)) 일 것 같습니다..!

Reply:
```swift
// Given
let number1 = [1,2,3,4]

// Then 
XCTAssertThrowsError(try UserBall(numbers: number1))
```
성우님 말처럼 위와 같이 될 것 같네요

Reply:
given when then 은 가독성을 향상시키기 위함입니다.

질문의 예시로 보여준 코드는 간단하고 단계도 따로 없어 가독성이 좋다고 판단됩니다. 이런 경우에 굳이 모든 단계를 지키려고 분리하지않아도 됩니다!